django/db/models/fields/__init__.py

```diff
@@ -1963,6 +1963,8 @@ def get_internal_type(self):
         return "GenericIPAddressField"
 
     def to_python(self, value):
+        if value and not isinstance(value, six.string_types):
```

Comment:
You've lost the `value = value.strip()` (line 1223 as of `master` at 8714403614c4dfa37e806db4a6708b8a91a827a4)